src/main/java/de/tum/in/www1/artemis/security/annotations/enforceRoleInExercise/EnforceRoleInExerciseAspect.java

```diff
@@ -0,0 +1,73 @@
+package de.tum.in.www1.artemis.security.annotations.enforceRoleInExercise;
+
+import static de.tum.in.www1.artemis.config.Constants.PROFILE_CORE;
+import static de.tum.in.www1.artemis.security.annotations.AnnotationUtils.getAnnotation;
+import static de.tum.in.www1.artemis.security.annotations.AnnotationUtils.getIdFromSignature;
+
+import java.util.Optional;
+
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Pointcut;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+import de.tum.in.www1.artemis.service.AuthorizationCheckService;
+
+@Profile(PROFILE_CORE)
+@Component
+@Aspect
+public class EnforceRoleInExerciseAspect {
```

Comment:
Would it be possible to use some common functions together with EnforceRoleInCourseAspect?
Especially for future extensions with more annotations for different resources